File: 20190519python/0941. Valid Mountain Array.py
class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        if len(A)<3:
            return False
        
        cur=0
        for k in range(len(A)-1):
            if A[k]==A[k+1]:
                return False
            if A[k]>A[k+1]:
                cur=k
                break
        if cur==0:
            return False
        for k in range(cur,len(A)-1):
            if A[k]==A[k+1]:
                return False
            if A[k]<A[k+1]:
                return False
        return True

File: archives/C#/0941. Valid Mountain Array.cs
public class Solution {
    public bool ValidMountainArray(int[] A) {
        if(A.Length<3){return false;}
        int top=0;
        for(int i=1;i<A.Length;i++){
            if(A[i]==A[i-1]){
                return false;
            }
            else if(A[i]<A[i-1]){
                top=i-1;
                break;
            }
        }
        if(top==0){return false;}
        for(int i=top+1;i<A.Length;i++){
            if(A[i]>=A[i-1]){
                return false;
            }
        }
        return true;
    }
}
